patterns/ecs-v1/exim

```diff
@@ -1,13 +1,25 @@
 EXIM_MSGID [0-9A-Za-z]{6}-[0-9A-Za-z]{6}-[0-9A-Za-z]{2}
-EXIM_FLAGS (<=|[-=>*]>|[*]{2}|==)
-EXIM_DATE %{YEAR:exim_year}-%{MONTHNUM:exim_month}-%{MONTHDAY:exim_day} %{TIME:exim_time}
-EXIM_PID \[%{POSINT}\]
+# <=     message arrival
+# =>     normal message delivery
+# ->     additional address in same delivery
+# *>     delivery suppressed by -N
+# **     delivery failed; address bounced
+# ==     delivery deferred; temporary problem
+EXIM_FLAGS (<=|=>|->|*>|**|==|<>|>>)
+EXIM_DATE (:?%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME})
+EXIM_PID \[%{POSINT:[process][pid]:int}\]
 EXIM_QT ((\d+y)?(\d+w)?(\d+d)?(\d+h)?(\d+m)?(\d+s)?)
 EXIM_EXCLUDE_TERMS (Message is frozen|(Start|End) queue run| Warning: | retry time not reached | no (IP address|host name) found for (IP address|host) | unexpected disconnection while reading SMTP command | no immediate delivery: |another process is handling this message)
-EXIM_REMOTE_HOST (H=(%{NOTSPACE:remote_hostname} )?(\(%{NOTSPACE:remote_heloname}\) )?\[%{IP:remote_host}\])
-EXIM_INTERFACE (I=\[%{IP:exim_interface}\](:%{NUMBER:exim_interface_port}))
-EXIM_PROTOCOL (P=%{NOTSPACE:protocol})
-EXIM_MSG_SIZE (S=%{NUMBER:exim_msg_size})
-EXIM_HEADER_ID (id=%{NOTSPACE:exim_header_id})
-EXIM_SUBJECT (T=%{QS:exim_subject})
+EXIM_REMOTE_HOST (H=(\[?%{NOTSPACE:[client][address]}\]? )?(\(%{NOTSPACE:[exim][log][remote_address]}\) )?\[%{IP:[client][ip]}\](?::%{POSINT:[client][port]:int})?)
```

Comment:
went with `client.address` here using `source.` sounded like there should be a `destination` field as well ... but maybe I am just overthinking - open to suggestions.

`[exim][log][remote_address]` is meant to be the "raw" unverified host from the smtp HELO record, exim than verifies this host and we end up with a `[client][address]` (optional) + the `[client][ip]`

Reply:
@kares You're correct, `source`/`destination` are typically populated together as a pair. But in an instance when only one of the pair is present, populating one without the other is fine. Likewise, the `client`/`server` fields are usually populated together as a pair.

If only the sender details are included, I'd lean towards populating `source` over `client` for these fields.

Reply:
thanks, have updated to use `source` / `destination`.